### ISSUE TITLE
Add Locale property for user to access in early data queries

### ISSUE THREAD

**Issue by riderjensen** [wpengine/faustjs#1228]
Discord Context: https://discord.com/channels/836253505944813629/877940107016019998/1064981832673808394

## Summary of problem:

In terms of data requests to get information on the page, we have the following: 
1. Seed query (gets passed to Component.variables)
2. Component.query (gets passed to props)
3. useQuery
4. useLazyQuery

Currently parts 1 and 2 happen before locale can be accessed (the soonest a locale can be accessed is on the props of a component, after a Component.query has run). This makes localization difficult since we would have to make another call with useQuery to get the data which runs client side or overfetch on the Components.query for all content in all locales and then run a filter on the data.

## Current Info

GetStaticPropsContext already has an optional locale property which users can add like so in their `_app.tsx`:

```
import '../faust.config';
import React from 'react';
import { useRouter } from 'next/router';
import type { AppProps } from 'next/app';
import { FaustProvider } from '@faustwp/core';

export default function MyApp({ Component, pageProps }: AppProps) {
  const router = useRouter();

  return (
    <FaustProvider pageProps={pageProps}>
      <Component {...pageProps} key={router.asPath} locale={router.locale} />
    </FaustProvider>
  );
}

```

This context is passed to `getWordPressProps` in the `getStaticProps` function. For example:
```
import { getWordPressProps, WordPressTemplate } from '@faustwp/core';
import type { GetStaticPropsContext } from 'next';
import type { PageProps } from '../types';

export default function Page(props: PageProps) {
  return <WordPressTemplate {...props} />;
}

export function getStaticProps(ctx: GetStaticPropsContext) {
  return getWordPressProps({ ctx });
}

export async function getStaticPaths() {
  return {
    paths: [],
    fallback: 'blocking',
  };
}

```

## Solution

Because `getWordPressProps` already has the context with locale on it, the solution I would propose is adding locale to one or both places listed here:

[Seed Query](https://github.com/wpengine/faustjs/blob/canary/packages/faustwp-core/src/getWordPressProps.tsx#L67)
[Component.Variables](https://github.com/wpengine/faustjs/blob/canary/packages/faustwp-core/src/getWordPressProps.tsx#L88)

Using it in the seed query would provide unique benefits to Faust's pluggable nature and allow users to write their custom seed queries with their own variables.

**Comment by youdianfan** [comment]
support

**Comment by josephfusco** [comment]
Closed by https://github.com/wpengine/faustjs/pull/1229 and is available in [@faustwp/core@0.2.4](https://www.npmjs.com/package/@faustwp/core).

Thank you for your work on this @riderjensen! 🎉
